platforms/jvm/testing-jvm/src/integTest/groovy/org/gradle/testing/testsuites/dependencies/TestSuitesGroovyDSLDependenciesIntegrationTest.groovy

```diff
@@ -1013,7 +1013,7 @@ class TestSuitesGroovyDSLDependenciesIntegrationTest extends AbstractIntegration
 
         expect:
         fails 'help'
-        result.assertHasErrorOutput("Could not find method getAt() for arguments [[org.apache.commons:commons-lang3:3.11, com.google.guava:guava:30.1.1-jre]] on DependencyAdder for ${suiteName}Implementation of type org.gradle.api.internal.artifacts.dsl.dependencies.DefaultDependencyAdder_Decorated.")
+        result.assertHasErrorOutput("Could not find method getAt() for arguments [[org.apache.commons:commons-lang3:3.11, com.google.guava:guava:30.1.1-jre]] on property 'dependencies.implementation' of type org.gradle.api.internal.artifacts.dsl.dependencies.DefaultDependencyCollector_Decorated.")
```

Comment:
🤔 I know we were already showing the internal name before, but is there some way to make this say `DependencyCollector` instead?  IIRC, `HasPublicType` might do it?

Reply:
Comes from here: https://github.com/gradle/gradle/blob/b0bde4fd33e97bd774f909b5ab79403313827deb/platforms/core-configuration/model-core/src/main/java/org/gradle/internal/instantiation/generator/AsmBackedClassGenerator.java#L882-L887

the single-arg `new BeanDynamicObject` constructor sets the publicType as null. We should probably change the codegen here too

Reply:
I think we can address this with the other thing above separately